Add Jenkins jobs for 3 bugzilla tasks

bkearney opened the pull request:
Add Jenkins jobs for 2 bugzilla tasks:
* pmscore - Runs nightly to rescore bugs based on caseload and bug criteria
* redmin-sync - Runs every 4 hours to sync bugzilla status from redmine issues

bkearney commented:
@SatelliteQE what is the review process?

lpramuk commented:
This repo is shared between QE and Devs
so @evgeni and @ekohl are good candidates to review this PR

evgeni commented:
Probably more @ehelms than Ewoud ;)

evgeni commented:
@bkearney go to our team irc and say "!review <url of this PR>" this will create a card on our board ;)

ekohl commented on workflows/bugzilla/aha-score.groovy:
```diff
@@ -0,0 +1,34 @@
+            ])
+        }
+
+        virtEnv('./ci', 'pip install psycopg2 python-bugzilla configparser future requests')
```
A common thing to do is list these in a file like `requirements.txt` and use `pip install -r requirements.txt`. That way you don't duplicate these between two repos (and multiple places in this repo).

bkearney replied:
Added this in.

ekohl commented on workflows/bugzilla/aha-score.groovy:
```diff
@@ -0,0 +1,34 @@
+    stage('Score AHA Cards') {
+        withCredentials([string(credentialsId: 'PM_AHA_ACCESS_TOKEN', variable: 'ACCESS_TOKEN')]) {
+
+          virtEnv('./ci', 'cd scripts && ./aha_scoring.py -d -r 7.0 --aha-token $ACCESS_TOKEN')
```
Probably a good practice to quote input
```suggestion
          virtEnv('./ci', 'cd scripts && ./aha_scoring.py -d -r 7.0 --aha-token "$ACCESS_TOKEN"')
```

bkearney replied:
added

bkearney commented on workflows/bugzilla/redmine-sync.groovy:
```diff
@@ -0,0 +1,35 @@
+                branches : [[name: 'master']],
+                extensions: [[$class: 'CleanCheckout']],
+                userRemoteConfigs: [
+                    [url: "https://gitlab.sat.engineering.redhat.com/satellite6/katello-team.git"]
```
done

bkearney commented:
@evgeni @ekohl applied the requests which you made. Specifically:
- renamed the jobs sat-bz-*
- Moved common setup and email logic to a central file.
- Added in env variables for the git logic
- moved python requirements to a text file in katello-team
- quoted command lines

I also removed the aha-score job which is no longer used.

evgeni commented on workflows/bugzilla/bugzilla.groovy:
```diff
@@ -0,0 +1,38 @@
+def virtEnv(path, command) {
```
We have an identically named function in https://github.com/theforeman/foreman-infra/blob/master/puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/virtualEnv.groovy (which we include in many jobs), which only differs in the Python version to yours.

Having the same name twice (even if they aren't imported today in the same job) sounds like a source of problems later.

Can we rename yours to `virtEnv3` or similar (and put it into a more generic file, ideally), or can you just use the one we already have (and thus use Python2)?

evgeni commented on workflows/bugzilla/bugzilla.groovy:
```diff
@@ -0,0 +1,38 @@
+  emailext(
+    subject: "${text}",
+    attachLog: true,
+    from: "pm-sat@redhat.com",
+    to: "bkearney@redhat.com"
```
ideally those are stored as variables in Jenkins, and you only refer to vars here.

see e.g. https://github.com/SatelliteQE/robottelo-ci/blob/0f6adfc14f009978ba96c4300d8c63468918fd1d/workflows/dogfoodCleanup.groovy#L27

evgeni commented on workflows/bugzilla/pmscore.groovy:
```diff
@@ -0,0 +1,19 @@
+                     usernameVariable: 'TEIID_USERNAME', passwordVariable: 'TEIID_PASSWORD'),
+                     string(credentialsId: 'PM_SAT_API_KEY', variable: 'API_KEY')]) {
+
+          virtEnv('./ci', 'cd scripts && ./bug_scoring.py --bz_username pm-sat@redhat.com --bz_api_key "$API_KEY" --teiid_username "$TEIID_USERNAME" --teiid_password "$TEIID_PASSWORD" --teiid_host virtualdb.engineering.redhat.com --product \'Red Hat Satellite\'')
```
```suggestion
          virtEnv('./ci', "cd scripts && ./bug_scoring.py --bz_username pm-sat@redhat.com --bz_api_key '${env.API_KEY}' --teiid_username '${env.TEIID_USERNAME}' --teiid_password '${env.TEIID_PASSWORD}' --teiid_host virtualdb.engineering.redhat.com --product 'Red Hat Satellite'")
```

This uses Groovy variable substitution instead of the shell environment, and eliminates the weird quoting that you needed before.

evgeni reviewed (commented):
Thanks @bkearney!

I left a few comments, but I don't think any of these should be blocking. Please ping and say if you want to address them now or we should merge this and address later.

bkearney commented:
@evgeni I made all the changes you requested except 1. I need to use python 3 since, you know, it has been out since 1990. I also kept it local to my file just to keep the number of files down. I am happy to move it out if we decide to move other apps to python3.

Changes made include using the emails from env, better use of single ticks, and renaming the command to be virtEnv3.

evgeni commented:
Thanks @bkearney!
